src/dsp_tools/utils/xmlupload/xmlupload.py

```diff
@@ -78,29 +78,37 @@ def xmlupload(
     )
 
     # establish connection to DSP server
-    con = login(server=server, user=user, password=password, dump=config.dump)
+    con = login(server=server, user=user, password=password, dump=config.diagnostics.dump)
     sipi_server = Sipi(sipi, con.get_token())
 
-    resources, permissions_lookup, resclass_name_2_type, stash = _prepare_upload(
+    resources, permissions_lookup, stash = _prepare_upload(
         root=root,
         con=con,
         default_ontology=default_ontology,
-        shortcode=shortcode,
-        verbose=config.verbose,
+        verbose=config.diagnostics.verbose,
     )
 
+    project_client: ProjectClient = ProjectClientLive(config.server, config.shortcode)
```

Comment:
mypy has issues with protocol classes, and at some point I was under the impression that this helps - not sure if it really does.  
But I actually like the type hints because they discourage from accidentally using the implementation type somewhere in the dependencies. But I can also remove them, if you prefer?

Reply:
What do you mean by "somewhere in the dependencies"? (I just wonder, the type annotation doesn't hurt me :-)

Reply:
I mean that this object will be passed to some method/function/constructor, and these methods should require the protocol, not the implementation. Does that make sense now?

Reply:
A now I see, that makes sense